### PR TITLE
Theme toggle & language toggle

### PR DESCRIPTION
**Pull request by bartosz-szkolnik**

### CODE REVIEW

File: src/app/[lang]/layout.tsx [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
-  children,
-}: Readonly<{
-  children: React.ReactNode;
-}>) {
+export default function RootLayout({ children }: Readonly<{ children: React.ReactNode }>) {
   return (
-    <html lang="en">
-      <body className={`${inriaSans.className} bg-gradient-to-r from-gradient-1 via-gradient-2 to-gradient-3`}>
-        {children}
+    <html lang="en" suppressHydrationWarning>
```
Comment (stasiakm) [inline src/app/[lang]/layout.tsx]: Probably `lang` should be dynamic

File: src/app/sections/hero.tsx [hunk 2]
```diff
@@ -1,20 +1,40 @@
...
-          {dict.TITLE}
-        </h1>
+      <div className="aria-hidden hidden md:block">
+        <div className="absolute left-1/2 top-[96px] -translate-x-2/4">
+          <h1
+            style={{ wordSpacing: '0.6rem', letterSpacing: '-0.05rem' }}
+            className={cn(
+              'border-0 border-b-4 border-b-chocolate text-center text-[6rem] font-extrabold text-darker-vanilla [-webkit-text-stroke:#363636_4.5px]',
+              heroFont.className,
+            )}
```
Comment (stasiakm) [inline src/app/sections/hero.tsx]:
```suggestion
            className={cn(
              'border-0 border-b-4 border-b-chocolate text-center text-[6rem] font-extrabold text-darker-vanilla [-webkit-text-stroke:#363636_4.5px] [word-spacing:0.6rem] [letter-spacing:-0.05rem]',
              heroFont.className,
            )}
```

File: src/app/sections/hero.tsx [hunk 3]
```diff
@@ -1,20 +1,40 @@
...
-          {dict.TITLE}
-        </h1>
+      <div className="block md:hidden">
+        <div className="absolute left-1/2 top-[96px] flex h-32 w-11/12 -translate-x-2/4 items-center justify-center rounded-xl bg-gradient-middle p-4 pt-2 dark:bg-chocolate">
+          <h1
+            style={{ wordSpacing: '0.25rem' }}
+            className={cn(
+              'border-4 border-gradient-middle border-b-chocolate pb-2 pt-0 text-center text-2xl font-semibold leading-normal text-chocolate dark:border-chocolate dark:border-b-vanilla dark:text-vanilla',
+              heroFont.className,
+            )}
```
Comment (stasiakm) [inline src/app/sections/hero.tsx]:
```suggestion
            className={cn(
              'border-4 border-gradient-middle border-b-chocolate pb-2 pt-0 text-center text-2xl font-semibold leading-normal text-chocolate dark:border-chocolate dark:border-b-vanilla dark:text-vanilla [word-spacing:0.25rem]',
              heroFont.className,
            )}
```

File: src/components/language-toggle.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            {supportsEmoji ? (
+              <>
+                {(locale === 'en' || locale === 'en-US' || locale === 'en-GB') && <span className="text-2xl">🇬🇧</span>}
+                {(locale === 'pl' || locale === 'pl-PL') && <span className="text-2xl">🇵🇱</span>}
+              </>
+            ) : locale === 'pl' || locale === 'pl-PL' ? (
+              <Icon variant="polish-flag" className="h-[1.2rem] w-[1.2rem]" />
+            ) : (
+              <Icon variant="great-britain-flag" className="h-[1.2rem] w-[1.2rem]" />
+            )}
```
Comment (stasiakm) [inline src/components/language-toggle.tsx]:
```tsx
function Flag({ supportsEmoji, locale }: { supportsEmoji: boolean; locale: Locale }) {
  switch (locale) {
    case 'en':
    case 'en-GB':
    case 'en-US': {
      if (!supportsEmoji) {
        return <Icon variant="great-britain-flag" className="h-[1.2rem] w-[1.2rem]" />;
      }

      return <span className="text-2xl">🇬🇧</span>;
    }
    case 'pl':
    case 'pl-PL': {
      if (!supportsEmoji) {
        return <Icon variant="polish-flag" className="h-[1.2rem] w-[1.2rem]" />;
      }

      return <span className="text-2xl">🇵🇱</span>;
    }
  }
}
```


```suggestion
            <Flag supportsEmoji={supportsEmoji} locale={locale} />
```

File: src/components/theme-toggle.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+'use client';
+
+import * as React from 'react';
```
Comment (stasiakm) [inline src/components/theme-toggle.tsx]:
```suggestion

```

File: src/components/ui/icon.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    >
+      {variant === 'settings' && ICONS['settings']}
+      {variant === 'polish-flag' && ICONS['polish-flag']}
+      {variant === 'great-britain-flag' && ICONS['great-britain-flag']}
+      {variant === 'custom' && children}
+    </div>
+  );
+}
+
+const ICONS = {
```
Comment (stasiakm) [inline src/components/ui/icon.tsx:26]: Consider using SVG Sprite

https://benadam.me/thoughts/react-svg-sprites/

File: src/components/language-toggle.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        </Button>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent align="end">
+        <DropdownMenuItem onClick={() => handleLocaleChange('en-GB')}>
+          <span className="flex items-center gap-2">
+            {supportsEmoji ? (
+              <span className="text-2xl">🇬🇧</span>
+            ) : (
+              <Icon variant="great-britain-flag" className="h-[1.2rem] w-[1.2rem]" />
+            )}
```
Comment (stasiakm) [inline src/components/language-toggle.tsx]:
```suggestion
            <Flag supportsEmoji={supportsEmoji} locale="en-GB" />
```

File: src/components/language-toggle.tsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            {dict.ENGLISH}
+          </span>
+        </DropdownMenuItem>
+        <DropdownMenuItem onClick={() => handleLocaleChange('pl-PL')}>
+          <span className="flex items-center gap-2">
+            {supportsEmoji ? (
+              <span className="text-2xl">🇵🇱</span>
+            ) : (
+              <Icon variant="polish-flag" className="h-[1.2rem] w-[1.2rem]" />
+            )}
```
Comment (stasiakm) [inline src/components/language-toggle.tsx]:
```suggestion
            <Flag supportsEmoji={supportsEmoji} locale="pl-PL" />
```

File: src/components/ui/icon.tsx (hunk 6 above)
Comment (bartosz-szkolnik) [reply src/components/ui/icon.tsx:26]: I will start using it with another update to the project. Thanks for the recommendation tho 😉

**Pull request merged**